File: SingleAgenda/SingleAgenda.Entities/Contact/Person.cs
using SingleAgenda.Entities.Base;
using SingleAgenda.Entities.Location;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace SingleAgenda.Entities.Contact
{
    public class Person
        : EntityBase
    {

        [Required]
        [Column(TypeName = "varchar(80)")]
        public string Name { get; set; }

        [Required]
        public PersonType PersonType { get; set; }

        [Required]
        [Column(TypeName = "varchar(50)")]
        public string Email { get; set; }

        /// <summary>
        /// The value of CPF / CNPJ
        /// </summary>
        [Column(TypeName = "varchar(14)")]
        public string Document { get; set; }

        public List<Address> Addresses { get; set; }

        #region Legal Person Specific Fields

        [Column(TypeName = "varchar(50)")]
        public string TradeName { get; set; }

        #endregion

        #region Natural Person Specific Fields

        public DateTime Birthday { get; set; }

        public Gender Gender { get; set; }

        #endregion

    }
}

File: web-portal/src/app/core/model/contacts/person.ts
import { BaseModel } from "../base/base-model";
import { Address } from "../location/address";
import { Gender } from "./gender";
import { PersonType } from "./person-type";

export class Person extends BaseModel {

    constructor(){
        super();
    }

    public name: string = "";
    public email: string = "";
    public personType: PersonType; 
    public document: string = "";
    public tradeName: string = "";
    public birthday: Date;
    public gender: Gender;
    public adresses: Address[] = [];

}